tap_iterable/iterable.py

```diff
@@ -137,11 +142,19 @@ def templates(self, column_name, bookmark):
       "InApp",
       "SMS"
     ]
+    # bookmark value `updatedAt` is getting round off to next second
+    # while putting into the STATE and which is causing 0 records fetched
+    # with existing bookmark value in STATE 
+    bookmark_val = utils.strptime_with_tz(bookmark)- timedelta(seconds=1)
```

Comment:
is this only a bug for templates stream?

Reply:
Yes, this is the bug with `templates` stream only.
Actually, I have added the millisecond level granularity to maintain the bookmark value.
So, before I was maintaining bookmark date like this "updatedAt": "2023-02-22T07:31:15.000000Z"
And, now, I'm storing it like "updatedAt": "2023-02-22T07:31:15.145000Z"
Still, the issue persists and it is actually API side bug that if we keep `startDateTime`= "2023-03-01+07%3A31%3A15"
It doesn't fetch the records with updatedAt= "2023-03-01 07:31:15" though record with that time present for the stream template. And, hence, subtracted 1 second from the existing bookmark value and made it to updatedAt= "2023-03-01 07:31:14" to fetch at least 1 record.